drivers/espi/host_subs_npcx.c

```diff
@@ -496,55 +543,31 @@ static void host_port80_isr(const void *arg)
 		}
 
 	} else {
-		uint16_t port80_buf[16];
-		uint8_t count = 0;
-		uint32_t code = 0;
-
-		while (IS_BIT_SET(inst_shm->DP80STS, NPCX_DP80STS_FNE) &&
-		       count < ARRAY_SIZE(port80_buf)) {
-			port80_buf[count++] = inst_shm->DP80BUF;
-		}
+		struct ring_buf *rbuf = &host_sub_data.port80_ring_buf;
 
-		for (uint8_t i = 0; i < count; i++) {
-			uint8_t offset;
-			uint32_t buf_data;
-
-			buf_data = port80_buf[i];
-			offset = GET_FIELD(buf_data, NPCX_DP80BUF_OFFS_FIELD);
-			code |= (buf_data & 0xFF) << (8 * offset);
-
-			if (i == count - 1) {
-				evt.evt_data = code;
-				espi_send_callbacks(host_sub_data.callbacks,
-						    host_sub_data.host_bus_dev, evt);
-				break;
-			}
+		while (IS_BIT_SET(inst_shm->DP80STS, NPCX_DP80STS_FNE)) {
+			struct npcx_dp80_buf dp80_buf;
 
-			/* peek the offset of the next byte */
-			buf_data = port80_buf[i + 1];
-			offset = GET_FIELD(buf_data, NPCX_DP80BUF_OFFS_FIELD);
-			/*
-			 * If the peeked next byte's offset is 0 means it is the start
-			 * of the new code. Pass the current code to Port80
-			 * common layer.
-			 */
-			if (offset == 0) {
-				evt.evt_data = code;
-				espi_send_callbacks(host_sub_data.callbacks,
-						    host_sub_data.host_bus_dev, evt);
-				code = 0;
-			}
+			dp80_buf.u16 = inst_shm->DP80BUF;
+			ring_buf_put(rbuf, &dp80_buf.u8[0], 2);
 		}
+		k_work_submit(&host_sub_data.work);
 	}
 
 	/* If FIFO is overflow, show error msg */
 	if (IS_BIT_SET(status, NPCX_DP80STS_FOR)) {
 		inst_shm->DP80STS |= BIT(NPCX_DP80STS_FOR);
-		LOG_ERR("Port80 FIFO Overflow!");
+		LOG_DBG("Port80 FIFO Overflow!");
```

Comment:
Is this change intentional? LOG_ERR sounds reasonable here.

Reply:
Yes, this is the intentional change. Please refer to the comment 19 and 20 in issue [b:259971621](https://partnerissuetracker.corp.google.com/issues/259971621) for the explanation,